src/main/java/com/autotune/analyzer/performanceProfiles/PerformanceProfileInterface/ResourceOptimizationOpenshiftImpl.java

```diff
@@ -77,73 +80,92 @@ public List<KruizeRecommendationEngine> getEngines() {
     }
 
     @Override
-    public void generateRecommendation(KruizeObject kruizeObject, ExperimentResultData experimentResultData) {
+    public void generateRecommendation(KruizeObject kruizeObject, List<ExperimentResultData> experimentResultDataList, Timestamp interval_start_time, Timestamp interval_end_time) throws Exception {
+        /*
+            To restrict the number of rows in the result set, the Load results operation involves locating the appropriate method and configuring the desired limitation.
+            It's important to note that in order for the Limit rows feature to function correctly,
+            the CreateExperiment API must adhere strictly to the trail settings' measurement duration and should not allow arbitrary values
+        */
+        String experiment_name = kruizeObject.getExperimentName();
+        int limitRows = (int) ((
+                KruizeConstants.RecommendationEngineConstants.DurationBasedEngine.DurationAmount.LONG_TERM_DURATION_DAYS *
+                        KruizeConstants.DateFormats.MINUTES_FOR_DAY)
+                / kruizeObject.getTrial_settings().getMeasurement_durationMinutes_inDouble());
```

Comment:
Should this be `limitRows = LONG_TERM_MINS / durationMins + 1`?

Reply:
ok I see that it is existing code moved inside the `generateRecommendation` method now